### PR TITLE
Small ffa fixes

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: core/arch/arm/kernel/thread_spmc.c [hunk 1]
```diff
@@ -653,7 +653,8 @@ static void handle_blocking_call(struct thread_smc_args *args,
 	default:
 		EMSG("Unhandled blocking service ID %#"PRIx32,
 		     (uint32_t)args->a3);
-		panic();
+		spmc_set_args(args, direct_resp_fid, swap_src_dst(args->a1), 0,
+			      TEE_ERROR_BAD_PARAMETERS, 0, 0);
```
Comment (etienne-lms) [inline core/arch/arm/kernel/thread_spmc.c:657]: I think it's rather an FFA code that is expected: `FFA_INVALID_PARAMETERS`?

Comment (jenswi-linaro) [reply core/arch/arm/kernel/thread_spmc.c:657]: That should be the case with `FFA_ERROR`, but this is an `FFA_DIRECT_RESPONSE` that carries an OP-TEE specific value.

Comment (etienne-lms) [reply core/arch/arm/kernel/thread_spmc.c:657]: Ok

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>`

**Comment by jenswi-linaro** [comment]
Tag applied

**Pull request merged**